internal/handlers/speaker_handler.go

```diff
@@ -0,0 +1,96 @@
+package handlers
+
+import (
+	"backend/internal/middleware"
+	"backend/internal/models"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"gorm.io/gorm"
+)
+
+type SpeakerHandler struct {
+	db *gorm.DB
+}
+
+func NewSpeakerHandler(db *gorm.DB) *SpeakerHandler {
+	return &SpeakerHandler{db: db}
+}
+
+func (h *SpeakerHandler) Register(r *gin.RouterGroup) {
+	speaker := r.Group("/speaker")
+	{
+		// Public endpoints (require auth)
+		speaker.GET("", h.List)
+		speaker.GET("/:id", h.Get)
+
+		// Admin-only endpoints
+		admin := speaker.Group("/admin")
```

Comment:
same as above: no need to group with sepearate base URL for admin routes, just make sure the middleware `AdminRequired()` is there for the POST, PUT and DELETE endpoints